pkg/services/alerting/notifiers/pagerduty.go

```diff
@@ -143,6 +158,20 @@ func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
 
 	body, _ := bodyJSON.MarshalJSON()
 
+	return body, nil
+}
+
+// Notify sends an alert notification to PagerDuty
+func (pn *PagerdutyNotifier) Notify(evalContext *alerting.EvalContext) error {
+	body, err := pn.BuildEventPayload(evalContext)
+	if err != nil {
+		pn.log.Error("Unable to build event pagerduty payload: ", err)
+	}
+	if evalContext.Rule.State == models.AlertStateOK && !pn.AutoResolve {
```

Comment:
Why not execute this before line 166, since `body` isn't used if this condition is true?

Reply:
good point